base/linalg/bidiag.jl

```diff
@@ -67,6 +67,10 @@ function convert{T}(::Type{Tridiagonal{T}}, A::Bidiagonal)
 end
 promote_rule{T,S}(::Type{Tridiagonal{T}}, ::Type{Bidiagonal{S}})=Tridiagonal{promote_type(T,S)}
 
+# Convert from Bidiagonal{Told} to Bidiagonal{Tnew} or AbstractMatrix{Tnew}(->Bidiagonal{Tnew})
+convert{Tnew,Told}(::Type{Bidiagonal{Tnew}}, A::Bidiagonal{Told}) = Bidiagonal(convert(Vector{Tnew}, A.dv), convert(Vector{Tnew}, A.ev), A.isupper)
+convert{Tnew,Told}(::Type{AbstractMatrix{Tnew}}, A::Bidiagonal{Told}) = convert(Bidiagonal{Tnew}, A)
```

Comment:
Which part? Those lines follow https://github.com/JuliaLang/julia/blob/9bab4da7dbee7631319e37a01f039f1d059e3eff/base/linalg/triangular.jl#L23-L24 which seem the same? Also, tests pass which are dependent on the functionality?